Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -778,9 +777,22 @@ private bool IsProjectGlobalDeclaration(Declaration candidateDeclaration, Declar
         {
             return candidateDeclaration.ProjectName == scopingDeclaration.ProjectName
                 && !(candidateDeclaration.ParentScopeDeclaration is ClassModuleDeclaration)
-                && (candidateDeclaration.Accessibility == Accessibility.Public
-                    || ((candidateDeclaration.Accessibility == Accessibility.Implicit)
-                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)));
+                && (IsExplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration)
+                || IsImplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration));
+        }
+
+        private bool IsExplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                    && candidateDeclaration.Accessibility == Accessibility.Public;
+        }
+
+        private bool IsImplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                        && candidateDeclaration.Accessibility == Accessibility.Implicit
+                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)
+                        && !candidateDeclaration.IdentifierName.StartsWith("Option ");
```

Comment:
Yes - the DeclarationType.ModuleOption was where I started, and basically, it was the source of the question I posted yesterday.  I could find no combination of flag/enum evaluation, casting, or conversion that provided a correct result.  That said, I see it being used other places in the code(!?).  So - I'm admittedly stumped by this one.